scripts/manage-db.py

```diff
@@ -130,22 +150,64 @@ def extract_active_data(url, dump_location='dump.sql'):
                OR releases.name = rules_scheduled_changes.base_mapping \
                OR releases.name = rules_scheduled_changes.base_whitelist \
             )" \
-        >> %s' % (mysql_default_command, database, dump_location,))
+        >> %s' % (mysql_default_command, database, dump_location))
     )
 
     popen(
         _strip_multiple_spaces('%s %s releases_history --where="releases_history.name=\'Firefox-mozilla-central-nightly-latest\' \
-        AND 1 limit 50" >> %s' % (mysql_default_command, database, dump_location,))
+        limit 50" >> %s' % (mysql_default_command, database, dump_location))
     )
 
     popen(
         _strip_multiple_spaces('%s --skip-add-drop-table --no-create-info %s releases_history --where "name = ( \
             SELECT rules.mapping \
             FROM rules \
             WHERE rules.alias=\'firefox-release\' \
-        ) LIMIT 50" >> %s' % (mysql_default_command, database, dump_location,))
+        ) LIMIT 50" >> %s' % (mysql_default_command, database, dump_location))
     )
 
+    query_release_mapping = """SELECT rules.mapping \
+            FROM rules \
+            WHERE rules.mapping IS NOT NULL \
+            UNION \
+            SELECT rules.fallbackMapping \
+            FROM rules \
+            WHERE rules.fallbackMapping IS NOT NULL \
+            UNION \
+            SELECT rules.whitelist \
+            FROM rules \
+            WHERE rules.whitelist IS NOT NULL"""
+
+    popen(_strip_multiple_spaces('''%s %s releases --where="releases.name IN (%s)" \
+        >> %s''' % (mysql_default_command, database, query_release_mapping, dump_location)))
+
+    query_release = """SELECT * \
+    FROM releases \
+    WHERE releases.name IN (%s)""" % query_release_mapping
+
+    result = trans.execute(query_release).fetchall()
+    partial_release_names = []
```

Comment:
You're going to end up with a lot of duplicates, this will be much easier if you use a set() (https://docs.python.org/2/library/sets.html).